packages/ui-components/src/components/JsonViewer/JsonViewer.component.tsx

```diff
@@ -0,0 +1,606 @@
+  ellipsis?: string
+  brace?: string
+  key?: string
+  index?: string
+  size?: string
+  border?: string
+  highlight: {
+    foreground?: string
+    background?: string
+  }
+  toolbar: {
+    border?: string
+    background?: string
+    foreground?: string
+  }
+  icon: {
+    expanded?: string
+    collapsed?: string
+    expandAll?: string
+  }
+  dataType: {
+    array?: string
+    object?: string
+    boolean?: string
+    date?: string
+    float?: string
+    function?: string
+    integer?: string
+    string?: string
+    nan?: string
+    null?: string
+    undefined?: string
+    regexp?: string
+    background?: string
+  }
+}
+
+interface ThemeContextType {
+  colors: ThemeColors
+  searchTerm?: string | null
+  truncate?: number | boolean
+  onExpandAll?: OnExapandFunc
+  onSearch?: OnSearchFunc
+
+  expanded: number | boolean
+  indentWidth: number
+  expandAll?: ExpandControlType | null
+}
+
+interface ExpandControlType {
+  expanded: boolean
+  timestamp: number
+}
+
+// eslint-disable-next-line no-unused-vars
+type OnExapandFunc = (value: boolean) => void
+// eslint-disable-next-line no-unused-vars
+type OnSearchFunc = (value: string | null) => void
+
+// Theme context to provide colors, ident size ect. in component tree
+const ThemeContext = React.createContext<ThemeContextType>({
+  colors: colorMap(DEFAULT_THEME),
+  indentWidth: DEFAULT_INDENT_WIDTH,
+  expanded: DEFAULT_EXPANDED,
+})
+
+interface ExpandIconProps {
+  expanded: boolean
+}
+// this component renders the expand icon depends on the expanded prop
+// per entry
+const ExpandIcon = ({ expanded }: ExpandIconProps) => {
+  const { colors } = useContext(ThemeContext)
+  return (
+    <svg
+      fill={expanded ? colors.icon.expanded : colors.icon.collapsed}
+      width="1em"
+      height="1em"
+      viewBox="0 0 1792 1792"
+      style={{
+        verticalAlign: "middle",
+        color: "var(--color-syntax-highlight-base0E)",
+        height: "1em",
+        width: "1em",
+      }}
+    >
+      <title>Expand/Collapse</title>
+      {expanded ? (
+        <path d="M1344 800v64q0 14-9 23t-23 9h-832q-14 0-23-9t-9-23v-64q0-14 9-23t23-9h832q14 0 23 9t9 23zm128 448v-832q0-66-47-113t-113-47h-832q-66 0-113 47t-47 113v832q0 66 47 113t113 47h832q66 0 113-47t47-113zm128-832v832q0 119-84.5 203.5t-203.5 84.5h-832q-119 0-203.5-84.5t-84.5-203.5v-832q0-119 84.5-203.5t203.5-84.5h832q119 0 203.5 84.5t84.5 203.5z"></path>
+      ) : (
+        <path d="M1344 800v64q0 14-9 23t-23 9h-352v352q0 14-9 23t-23 9h-64q-14 0-23-9t-9-23v-352h-352q-14 0-23-9t-9-23v-64q0-14 9-23t23-9h352v-352q0-14 9-23t23-9h64q14 0 23 9t9 23v352h352q14 0 23 9t9 23zm128 448v-832q0-66-47-113t-113-47h-832q-66 0-113 47t-47 113v832q0 66 47 113t113 47h832q66 0 113-47t47-113zm128-832v832q0 119-84.5 203.5t-203.5 84.5h-832q-119 0-203.5-84.5t-84.5-203.5v-832q0-119 84.5-203.5t203.5-84.5h832q119 0 203.5 84.5t84.5 203.5z"></path>
+      )}
+    </svg>
+  )
+}
+
+// Toolbar Icon
+const ExpandAllIcon = () => {
```

Comment:
I would extract this one and other one to nearby components but exported them only to make use inside of this component